src/main/kotlin/org/jitsi/nlj/RtpEncodingDesc.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.nlj
+
+import org.jitsi.nlj.rtp.SsrcAssociationType
+import org.jitsi.nlj.rtp.VideoRtpPacket
+import org.jitsi.nlj.stats.NodeStatsBlock
+
+/**
+ * Keeps track of information specific to an encoded RTP stream.
+ *
+ * @author Jonathan Lennox
+ */
+class RtpEncodingDesc
+@JvmOverloads
+constructor(
+    /**
+     * The primary SSRC for this encoding.
+     */
+    val primarySSRC: Long,
+    /**
+     * The [RtpLayerDesc]s describing the encoding's layers.
+     */
+    initialLayers: Array<RtpLayerDesc> = arrayOf()
+) {
+    /**
+     * The ssrcs associated with this encoding (for example, RTX or FLEXFEC)
+     * Maps ssrc -> type [SsrcAssociationType] (rtx, etc.)
+     */
+    private val secondarySsrcs: MutableMap<Long, SsrcAssociationType> = HashMap()
+
+    fun addSecondarySsrc(ssrc: Long, type: SsrcAssociationType) {
+        secondarySsrcs[ssrc] = type
+    }
+
+    internal var layers = initialLayers
+        set(newLayers) {
+            for (newLayer in newLayers) {
+                /* TODO: indices should be in order, so traverse old and new layers in linear not quadratic time? */
```

Comment:
Does this need to be addressed?

Reply:
Doesn't need to, but would be nice.  I couldn't easily figure out if there was idiomatic Kotlin for this, though.